http.go

```diff
@@ -152,3 +152,14 @@ func (ln tcpKeepAliveListener) Accept() (c net.Conn, err error) {
 	tc.SetKeepAlivePeriod(3 * time.Minute)
 	return tc, nil
 }
+
+func redirectToHTTPS(opts *Options, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		proto := r.Header.Get("X-Forwarded-Proto")
+		if opts.ForceHTTPS && r.TLS == nil && strings.ToLower(proto) != "https" {
```

Comment:
What happens here if the request to the upstream proxy is made via http, but then the upstream proxy makes a request to the proxy with https? I think we might want to change the condition here
```suggestion
		if opts.ForceHTTPS && (r.TLS == nil || (proto != "" && strings.ToLower(proto) != "https")) {
```
My thinking being that `x-Forwarded-Proto` could be `http` even though the actual request to the OAuth2 Proxy is made via `https` 🤔

Reply:
Good point. I'll accept the change.